### DISCUSSION TITLE
Possible to manipulate Livewire property from a third party JS library with no DOM element?

### DISCUSSION THREAD

**Discussion by rich06** [livewire/livewire#6884] [1. Help] [answered]
I am new to Livewire so forgive me if this is obvious but I searched the docs without really finding the answer... I have a Livewire component that needs to track some variables that are updated by an external .js module. These variables are calculated and are not part of the DOM but the values need to be sent back to the server from the client and I was hoping to use LW instead of using ajax calls from the .js script.

Is there a way to achieve this? It seems that LW needs a DOM element to hook into - is that correct?
Thanks!

**Comment by N1ebieski** [comment]
You can [access to the livewire component](https://livewire.laravel.com/docs/javascript#accessing-components) and then you'll be able to set a property on the component by name:

```js
// Set a property on the component by name...
// Usage: $wire.$set('count', 5)
$set(name, value, live = true) { ... },
```

**Reply by rich06** [reply]
Thanks for the response - as a quick test I added the following to the .js file

```
let components = Livewire.all()
console.log(components)

```
However it returns an empty array... do I need to register the component or is there something I have missed here?

**Comment by h-avatar** [comment]
You can using this method:
Livewire.getByName('componentName').set('var','value',live = true);

If you are using jquery and you are access to child element in component can using my custom function to set value. Useful when you have multiple identical components on the page.
‍‍‍‍
```js
function set_livewire_vars(el, name, value, live = false) {
    if (name.toString().trim() === '') {
        return false;
    }

    let jel;
    if (el instanceof jQuery) {
        jel = el.parents();
    } else {
        jel = $(el).parents();
    }

    if (jel.length) {
        for (let i = 0; i < jel.length; i++) {
            var attr = $(jel[i]).attr('wire:id');
            if (typeof attr !== 'undefined' && attr !== false) {
                Livewire.find(attr).set(name, value, live);
                return true;
            }
        }
    }

    return false;
}   
```

**Reply by rich06** [reply]
Thanks for the response. Livewire does not seem to have my component registered (see my response above) as it returns an empty array for Livewire.all() so I must be missing something here...?

**Reply by h-avatar** [reply]
If you add a livewire component to the page, then livewire should register it automatically. There is no need to do anything.
<livewire:componentName />

**Reply by rich06** [reply]
Okay thanks - the component is being loaded by the Filament PHP framework so maybe I need to find out why there is no LW component being registered by their code.

**Reply by rich06** [reply]
Some progress ... !

I have just tested without using the Filament framework. I implemented the Livewire quickstart counter demo from here (https://livewire.laravel.com/docs/quickstart). I spotted that LW issues an event when it has initialised so my .js code now runs when that event is emitted and I now see data returned by the `Livewire.all()` call. 

I then run the following (my component is named rich-test' and get:-
 
```
let test = Livewire.getByName('rich-test') 
console.log(test) // returns -> Array [Proxy]

test.set('count',42);
Error - Uncaught TypeError: test.set is not a function
```
Anything I have missed here?

**Reply by h-avatar** [reply, marked as answer]
Return value is Array. You must use with index like this:
```
test[0].set('count',42)
```

**Reply by rich06** [reply]
Many thanks @h-avatar that did it.. 👍
